openid-4-verifiable-presentations-1_0.md

```diff
@@ -549,11 +555,276 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+
+The Verifiable Presentation Query Language (VP Query) is a JSON-encoded query
+language that allows the Verifier to request the Wallet to present Verifiable
+Credentials that match the query. The Verifier MAY encode constraints on the
+combinations of credentials and claims that are requested. The Wallet evaluates the
+query against the Verifiable Credentials it holds and returns the Verifiable
+Presentations that match the query.
+
+A valid VP Query is defined as a JSON-encoded object with the following
+top-level properties:
+
+`expect_credentials`:
+: REQUIRED. A non-empty array of Credential Queries as defined in (#credential_query)
+that specify the requested Verifiable Credentials.
+
+`valid_credential_sets`:
+: OPTIONAL. A non-empty array containing arrays of identifiers for Credential Queries defined in
+`expect_credentials` that defines which sets of Credentials may be returned.
+The identifier MAY be postfixed by `?`, indicating that
+delivery of the the respective Credential is optional. For details, see
+(#vp_query_lang_processing_rules).
+
+Note: While this specification does not define additional top-level properties,
+future extensions MAY define additional properties that modify the behavior of
+the Wallet when processing the query.
+
+## Credential Query {#credential_query}
+
+A Credential Query is an object representing a request for one specific
+Credential. 
+
+It contains the following properties:
+
+`id`:
+: REQUIRED. A string identifying the Credential in the response and, if provided,
+the constraints in `valid_credential_sets`. The value MUST be a non-empty string
+consisting of alphanumeric, underscore (`_`) or hyphen (`-`) characters.
+Within the Authorization Request, the same `id` MUST NOT
+be present more than once.
+
+`format`:
+: REQUIRED. A string that specifies the format of the requested
+Verifiable Credential. Valid Credential Format Identifier values are defined in
+Appendix A of [@!OpenID.VCI]. The value of this property MUST be one of the
+supported Credential Formats as defined in the Wallet's metadata.
+
+`expect_meta`: 
+: OPTIONAL. An object defining additional properties requested by the Verifier that
+apply to the metadata and validity data of the Credential. The properties of
+this object are defined per Credential Format in (#format_specific_properties). If omitted,
+no specific constraints are placed on the metadata or validity of the requested
+Credential.
+
+`expect_claims`:
+: OPTIONAL. A non-empty array of objects as defined in (#claims_query) that specifies
+claims in the requested Credentials.
+
+`valid_claim_sets`:
+: OPTIONAL. A non-empty array containing arrays of identifiers for elements in
+`expect_claims`. The identifier MAY be postfixed by `?`, indicating that
+delivery of the the respective claim is OPTIONAL.  
+
+## Claims Query {#claims_query}
+
+Each entry in `expect_claims` MUST be an object with the following properties:
+
+`id`:
```

Comment:
Same comment as earlier about removing the limitations on the content of the string.

Reply:
please open a separate issue if this is still a concern.